recipes/libcurl/all/conanfile.py

```diff
@@ -396,11 +387,13 @@ def _generate_with_autotools(self):
             f"--enable-symbol-hiding={self._yes_no(self.options.with_symbol_hiding)}",
             f"--enable-unix-sockets={self._yes_no(self.options.with_unix_sockets)}",
         ])
+
         if self.options.with_ssl == "openssl":
             path = unix_path(self, self.dependencies["openssl"].package_folder)
             tc.configure_args.append(f"--with-ssl={path}")
-        else:
+        elif not self.options.with_ssl:
```

Comment:
no, revert please

Reply:
relevant commits to analyse differences:
https://github.com/curl/curl/commit/576e507c78bdd2ec88da442a5354f9dd661b9a8a
https://github.com/curl/curl/commit/7d69924ce773f13c175e56ad1b20b0d5dac224b9

Reply:
Since with-ssl has been deprecated in favor of with-openssl in 7.77.0 (https://github.com/curl/curl/commit/7bdec2a08bf025d2f66c168111d47df6b21890d9), I would replace openssl logic in autotools build by:

```python
        openssl_option = "ssl" if Version(self.version) < "7.77.0" else "openssl
        if self.options.with_ssl == "openssl":
            path = unix_path(self, self.dependencies["openssl"].package_folder)
            tc.configure_args.append(f"--with-{openssl_option}={path}")
        elif not self.options.with_ssl:
            tc.configure_args.append(f"--without-{openssl_option}")
```